### PR TITLE
Port 4619 cancel resync event before starting a new one

### PR DESCRIPTION
**Pull request by yairsimantov20**
# Description

What - Aborting an already running resync when the resync is triggered
Why - Reync might conflict with each other and the framework might encounter a situation when a resync would like to start when there is already another one running
How - Using PyDispatcher and the asyncio task cancel im canceling the task when there is a resync event

## Type of change

- [X] New feature (non-breaking change which adds functionality)

a

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -54,6 +54,7 @@ click = { version = "^8.1.3", optional = true }
 rich = { version = "^13.4.1", optional = true }
 cookiecutter = { version = "^2.1.1", optional = true }
 jinja2-time = { version = "^0.2.0", optional = true }
+pydispatcher = "^2.0.7"
```
Comment (Tankilevitch) [inline pyproject.toml:57]: seems like  it wasn't done using poetry, we need to have the .lock file modified as well

File: port_ocean/context/resource.py
Comment (Tankilevitch) [inline port_ocean/context/resource.py:1]: I suggest adding a comment in this file as well for what is needed for

File: port_ocean/context/event.py [hunk 2]
```diff
@@ -28,11 +44,39 @@ class EventType:
 class EventContext:
     event_type: str
     trigger_type: TriggerType = "machine"
-    resource_config: Optional["ResourceConfig"] = None
     attributes: dict[str, Any] = field(default_factory=dict)
+    _aborted: bool = False
     _port_app_config: Optional["PortAppConfig"] = None
     _parent_event: Optional["EventContext"] = None
     _event_id: str = field(default_factory=lambda: str(uuid4()))
+    _on_abort_callbacks: list[AbortCallbackFunction] = field(default_factory=list)
```
Comment (Tankilevitch) [inline port_ocean/context/event.py:52]: This class starts to maintain multiple configurations and scenarios might be worth adding documentation to the class

**Review by Tankilevitch — APPROVED** [review]
Nice one :) 
Minor docs commenting

Also I wonder whether we want to use that package as it doesn't seem much maintained and there is no adoption to it from what I see in github. Let me know what you think

File: pyproject.toml (hunk 1 above)
Comment (yairsimantov20) [reply pyproject.toml:57]: there is obe

File: docs/framework-guides/docs/_common/resync-abort-message.mdx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+:::danger
+Triggering your integration resync while there is an already running resync will abort the current resync and start a
```
Comment (MPTG94) [inline docs/framework-guides/docs/_common/resync-abort-message.mdx:2]:
```suggestion
If a resync event is received by your integration while it is actively performing a resync, the currently running resync will be aborted and a new resync process will start.
```

File: docs/framework-guides/docs/_common/resync-abort-message.mdx [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+new one.
```
Comment (MPTG94) [inline docs/framework-guides/docs/_common/resync-abort-message.mdx:3]:
```suggestion
```

File: docs/framework-guides/docs/_common/resync-abort-message.mdx [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+
+If this would repeatedly happen, it could cause your integration to never finish the resync process.
```
Comment (MPTG94) [inline docs/framework-guides/docs/_common/resync-abort-message.mdx:5]:
```suggestion
If a new resync trigger consistently aborts a running resync, it means that your integration never finishes a complete resync process (which means some information from the 3rd-party system might never appear in Port).
```

**Review by MPTG94 — APPROVED** [review]
Reviewed the docs, skimmed over the code, LGTM

**Pull request merged**